src/job/MomoJobBuilder.ts

```diff
@@ -0,0 +1,45 @@
+import { Handler, MomoJob } from './MomoJob';
+
+export class MomoJobBuilder implements Partial<MomoJob> {
+  name?: string;
+  interval?: string;
+  immediate?: boolean;
+  concurrency?: number;
+  maxRunning?: number;
+  handler?: Handler;
+
+  withName(name: string): this & Required<Pick<MomoJob, 'name'>> {
+    return Object.assign(this, { name: name });
```

Comment:
Can we set the field here on the partial and then return the complete builder
```typescript
this.job.name = name;
return this;
```

Reply:
We can set the field instead of using `Object.assign`. This is a good idea, since the use of `Object.assign` is just a trick to get around the type checker. Instead, we can perform an explicit cast in the return statement.

Returning these complex types allows us to enforce the validation through the type-system, which would offer typescript-users to find validation errors at compile-time. I think this feature is worth the extra complexity, and would like to keep it, if that's ok.

Reply:
I haven't seen this kind of pattern before, but I like the idea so I am fine with keeping that. I think the cast is the clearer approach here (actually wondering if the `Object.assign` way is still going to work once we finished our linting setup).